main.cpp

```diff
@@ -113,15 +97,17 @@ int main()
     T t2( 7, "Seven" );                                             
     
     Comparison f;                                            
-    auto* smaller = f.compare( &t1 , &t2 );                              
+    auto* smaller = f.compare( t1 , t2 );                           
     
     if( smaller != nullptr )
         std::cout << "the smaller one is << " << smaller->name << std::endl;
-    
+    else
+        std::cout << "a nullptr was passed to compare() resulting in compare() returning nullptr" << std::endl;
```

Comment:
What is the reason a nullptr would be returned?
It is not possible to pass pointers to `compare` in the current implementation
Your std::cout statement is incorrect as a result of this fact.
- [ ] revise